src/reflect/scala/reflect/internal/util/ThreadLocalStats.java

```diff
@@ -0,0 +1,39 @@
+package scala.reflect.internal.util;
+
+import scala.reflect.internal.util.Statistics;
+
+public class ThreadLocalStats {
```

Comment:
Why is this class defined in Java?

Reply:
Because of statistcs :smile:.

Reply:
I don't understand..

Reply:
It's a Java class so that I can define `static` members. They are necessary so that the same instance is accessible from both `IOStats` and the `AbstractFile` and `Path` definitions.

Reply:
Wouldn't a Scala `object` work?

Reply:
Yes, but wouldn't the `static` members be faster?

Reply:
Answering that question would require a benchmark. But by default we should write our code in Scala, unless a performance issue can be measured.

Reply:
In my previous PR, the difference between accessing a `static` member and a normal `object` member was quite drastic re performance, this is why here I defaulted to `static`.

Reply:
Can you point me at the benchmark?

Reply:
Maybe you're thinking about the special case of `MethodHandle`-s stored in final static vals. Invocations over these are inlinable by JIT.

Reply:
Benchmarks [here](https://scala-ci.typesafe.com/grafana/dashboard/db/pr-validation?var-scalaSha=311a576030|bb3e997faa|6d006a5461|63953e1388|5254231683|b5a3aee4db&var-source=All&var-bench=HotScalacBenchmark.compile&var-host=scalabench@scalabench@&from=1502073067834&to=1502882030576).
  
My first attempt for storing method handles [was modeled with an object](https://github.com/scalacenter/scala/blob/dbb772051fdc4b05af6f36c8f4501e6d96cd705b/src/reflect/scala/reflect/internal/util/Statistics.scala#L8-L11) and that made things slower, whereas using a static defined in Java made it significantly faster. This may be specific to method handles, though. However, I decided to follow the same logic in here.
  
I'll remove the cache and measure this concrete scenario.

Reply:
> Maybe you're thinking about the special case of MethodHandle-s stored in final static vals. Invocations over these are inlinable by JIT.
  
Yeah. That was it.